plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -103,4 +112,224 @@ public void testPropertiesTable()
             assertUpdate("DROP TABLE IF EXISTS " + tableName);
         }
     }
+
+    @Test
+    public void testPartitionsTable()
+    {
+        String tableName = "test_simple_partitions_table";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE) WITH (partitioned_by = ARRAY['_date'] )");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE)), (1, CAST('2019-09-09' AS DATE)), (2, CAST('2019-09-09' AS DATE))", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE)), (4, CAST('2019-09-10' AS DATE)), (5, CAST('2019-09-10' AS DATE))", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (6, NULL)", 1);
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 7");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 4");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('partition', 'row(_date date)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', ''),
+                            ('data', 'row(_bigint row(min bigint, max bigint, null_count bigint))', '', '')
+                            """);
+
+            assertQuery(
+                "SELECT partition._date FROM \"" + tableName + "$partitions\"", " VALUES DATE '2019-09-08', DATE '2019-09-09', DATE '2019-09-10', NULL");
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
```

Comment:
Use as well a test from https://github.com/trinodb/trino/tree/master/plugin/trino-delta-lake/src/test/resources/databricks73/no_column_stats when the column stats are missing.

On a second thought, you'll likely need to manually create a resource directory based on the previously mentioned one which would correspond to a table with partitions.

You can also add some data in the test with the current version of Trino which will render the table to have `add` mixed entries in a mixed state - with and without stats. Relevant for testing the accuracy of `data` column.

Reply:
Added `testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeSystemTableCompatibility#testTablePartitionsWithNoColumnStats`